cumulusci/robotframework/utils.py

```diff
@@ -199,3 +200,35 @@ def run_with_retry(self, *args, **kwargs):
     else:
         # Decorator was used without arguments
         return decorate(target)
+
+
+class PerfJSONConverter:
+    def __init__(self, jsondata):
+        self.data = json.loads(jsondata)
+
+    def to_dict(self, include_raw=False):
+        rc = {
+            metric["metrics"] + "-" + metricType: metric[metricType]
+            for metricType in ("totalTime", "totalCalls")
+            for metric in self.data["summary"]
+        }
+        if include_raw:
+            rc["_raw"] = self.data
+        return rc
+
+    def to_csv(self):
+        """Helper function to generate CSV-like data from performance metrics JSON"""
```

Comment:
But not actually CSV? What's the use case for this output?

Reply:
Well at the risk of drawing your eyes to some other questionable code I'll link here:

https://github.com/SFDO-Tooling/CumulusCI/pull/1135/files#diff-d58bc12615af74abeefd97ffb7ccf87fR261

It may be used as a Perf debugging tool and it MAY end up being the way I get the Perf data into MetaCI. Whether to have MetaCI read CSV-like lines out of output.xml OR try to weave together information from output.xml and perf.json is an open question right now. There are trade-offs either way.

In the April version of this prototype I put the perf data right into output.xml.

Regardless of how MetaCI reads it, I do think that having the JSON option is good, especially for the more detailed output modes.

Reply:
I think if this function returns something csv-like but not actually csv, maybe a name change is in order. Or perhaps even better, why not use a csv library to generate real csv?